libraries/chain/market_evaluator.cpp

```diff
@@ -205,10 +205,13 @@ object_id_type call_order_update_evaluator::do_apply(const call_order_update_ope
 
    if( o.delta_debt.amount != 0 )
    {
+      FC_ASSERT( _dynamic_data_obj->current_supply + o.delta_debt.amount <= GRAPHENE_MAX_SHARE_SUPPLY );
+      FC_ASSERT( _dynamic_data_obj->current_supply + o.delta_debt.amount >= 0 );
+
       d.adjust_balance( o.funding_account, o.delta_debt );
 
       // Deduct the debt paid from the total supply of the debt asset.
-      d.modify(*_dynamic_data_obj, [&](asset_dynamic_data_object& dynamic_asset) {
+      d.modify( *_dynamic_data_obj, [ this, &o ]( asset_dynamic_data_object& dynamic_asset ) {
```

Comment:
The ASSERTs were originally within the d.modify. Moved to before d.adjust_balance, as it seems to make more sense. Please correct me if the logic is flawed.

Also, some branches do not contain the verification that the current supply is non-negative. Perhaps that check is done elsewhere?